src/main/java/cr0s/warpdrive/config/structures/Schematic.java

```diff
@@ -65,28 +102,142 @@ public boolean loadFromXmlElement(final Element element) throws InvalidXmlExcept
 			} catch (final InvalidXmlException exception) {
 				exception.printStackTrace(WarpDrive.printStreamError);
 				WarpDrive.logger.error(String.format("Skipping invalid insertion %s",
-				                                     blockState));
+				blockState));
 			}
 		}
 		
 		return true;
 	}
 	
-	@Override
-	public boolean generate(@Nonnull final World world, @Nonnull final Random random, @Nonnull final BlockPos blockPos) {
-		return instantiate(random).generate(world, random, blockPos);
-	}
+	static class BlockMatcher {
+		
+		Block block;
+		IBlockState blockState;
+		boolean blockStateWildcard;
+		
+		public static BlockMatcher fromBlockStateString(String blockStateString) {
+			///TODO: allow different data input type for meta: range (e.g. 2-13), comma sepearated list (e.g. 1,2,3..), multiple property (e.g. variant=oak,half=bottom)
+			
+			BlockMatcher result = new BlockMatcher();
+			
+			String state = "";
+			String metaString = "*";
+			if (blockStateString.contains("@")) {    // with meta
+				String[] blockStateParts = blockStateString.split("@");
+				state = blockStateParts[0].trim();
+				metaString = blockStateParts[1].trim();
+			} else {    // without meta
+				state = blockStateString;
+			}
+			result.block = Block.getBlockFromName(state);
+			
+			if (result.block == null) {
+				WarpDrive.logger.warn(String.format("Found invalid block in replacement or insertion: %s", state));
+				return null;
+			}
+			if (metaString.equals("*")) {    //wildcard
+				result.blockStateWildcard = true;
+			} else if (metaString.matches(".*=.*")) {    // in string format (e.g. "color=red")
+				String[] metaParts = metaString.split("=");
+				String propertyKey = metaParts[0].trim();
+				String propertyValue = metaParts[1].trim();
+				/*
+				IProperty<? extends Comparable<?>> property = this.block.getBlockState().getProperty(propertyKey);
+				if(property == null){
+					WarpDrive.logger.warn(String.format("Found invalid block property %s for block %s", propertyKey, state));
+				}else {
+					Optional<? extends Comparable<?>> result = property.parseValue(propertyValue);
+					if (result.isPresent()){
+						this.blockState = this.block.getDefaultState().withProperty(property, result.get());
+					}else{
+						WarpDrive.logger.warn(String.format("Value %s is not allowed for property %s for block %s", propertyValue, propertyKey, state));
+					}
+				}
+				*/
+				// we tried.
```

Comment:
* Only the else content is different, consider trimming the rest?
* What is the actual issue encountered here? consider updating with a proper comment describing it?